### ISSUE TITLE
Directions API. Url provider 

### ISSUE THREAD

**Issue by RingerJK** [mapbox/mapbox-java#1129]
### Description
In terms of `Navigation for Android` exists 2 types of `Navigation`([Router](https://github.com/mapbox/mapbox-navigation-android/blob/d45035512a/libnavigation-base/src/main/java/com/mapbox/navigation/base/route/Router.kt)), it's `Offline` and `Online`. Both of them needs `Route-URL` provider.(`Offline` and `Online` are different _modules_ ,  have own dependency graph and might be used independent).

`Online Router`'s `Route URL` is provided by [MapboxDirections](https://github.com/mapbox/mapbox-java/blob/master/services-directions/src/main/java/com/mapbox/api/directions/v5/MapboxDirections.java) that is a part of `services-core` library(under the hood is _Retrofit_(+_OkHttp_) library)

### Problem 
`Offline Router` cannot consume _url_ from `services-core` because it brings "network" dependencies like _Retrofit_ and _OkHttp_(they shouldn't be in `Offline` navigation).
`Offline Router` has own `route-url` provider [RouteUrl](https://github.com/mapbox/mapbox-navigation-android/blob/d45035512a174ad2609f7d22d8c6eb95553e2e79/libnavigation-base/src/main/java/com/mapbox/navigation/base/route/internal/RouteUrl.kt)
The main issue here: two classes where the same piece of logic exist. 

### Solution 
- keep `service-core` library interfaces as it is
- create additional library (like `router-url-provider`) that depends on `service-directions-model` and (if java doesn't have clever approach for creating url) additional java library 

**How dependency tree looks now** 
<img width="453" alt="Screen Shot 2020-03-12 at 16 38 40" src="https://user-images.githubusercontent.com/8738058/76527814-e0e64400-6480-11ea-9c00-a36be6a55dc3.png">

**Suggestion**
<img width="639" alt="Screen Shot 2020-03-12 at 16 47 59" src="https://user-images.githubusercontent.com/8738058/76528060-463a3500-6481-11ea-9abe-dc7a51da0add.png">

### Pros
`router-url-provider` might be used independently on `service-core` 👍 

**cc** @mapbox/navigation-android @mapbox/maps-android @mapbox/navigation-api

**Comment by zugaldia** [comment]
Tagging @langsmith for initial review - you probably are the most familiar with the current set up.

**Comment by langsmith** [comment]
> Offline Router cannot consume url from services-core because it brings "network" dependencies like Retrofit and OkHttp(they shouldn't be in Offline navigation).
Offline Router has own route-url provider RouteUrl
The main issue here: two classes where the same piece of logic exist.



@RingerJK, is this still an issue? I ask because maybe things have changed in https://github.com/mapbox/mapbox-navigation-android since you opened this ticket. If so, is it an absolute blocker for navigation work?

**Comment by RingerJK** [comment]
@langsmith yes, it's still actual issue. It's not a blocker, because we have opportunity to get url for Offline routing via internal implementation(`RouteUrl`), but would be great to resolve current issue to avoid problems with url in future

**Comment by RingerJK** [comment]
@langsmith @zugaldia I can take it if you're never mind

**Comment by zugaldia** [comment]
@RingerJK should we address this issue for 1.0, or can I wait until after?

**Comment by RingerJK** [comment]
@zugaldia It's not a blocker for 1.0, event for 1.1. This Issue regarding resolve duplicating code in Navigation SDK and Directions API. 
Of course we might wait till 1.0 👍
